backend/packages/Upgrade/src/api/services/ExperimentService.ts

```diff
@@ -825,24 +816,24 @@ export class ExperimentService {
       if (promiseArray.length) {
         const metricsDocs = await Promise.all([...promiseArray]);
         queryDocsToSave = queryDocsToSave.map((queryDoc, index) => {
-          queryDoc.query.metric = metricsDocs[index];
+          metricsDocs ? queryDoc.metric = metricsDocs[index]: queryDoc.metric = null;
           return queryDoc;
         });
       }
       
       // saving conditions and saving partitions
       let conditionDocs: ExperimentCondition[];
       let partitionDocs: ExperimentPartition[];
-      let queryDocs: Query[];
+      let queryDocs: any;
       try {
         [conditionDocs, partitionDocs] = await Promise.all([
           this.experimentConditionRepository.insertConditions(conditionDocsToSave, transactionalEntityManager),
           this.experimentPartitionRepository.insertPartitions(partitionDocsToSave, transactionalEntityManager),
         ]);
-        if (queryDocsToSave) {
-          queryDocs = await Promise.all([this.queryRepository.insertQueries(queryDocsToSave, transactionalEntityManager)]);
+        if (queryDocsToSave.length > 0) {
```

Comment:
You should add this code in the above await statement with Promise.all else they will run sequentially

Reply:
okay. I tried that as you mentioned earlier, but I was not able to put condition on this queryDoc.